hivemq-edge/src/frontend/src/components/Chakra/ContextMenu.tsx

```diff
@@ -0,0 +1,79 @@
+import { MutableRefObject, useCallback, useEffect, useRef, useState } from 'react'
+import { useEventListener, Portal, Menu, MenuButton, PortalProps, MenuButtonProps, MenuProps } from '@chakra-ui/react'
+
+export interface ContextMenuProps<T extends HTMLElement> {
+  renderMenu: () => JSX.Element | null
+  children: (ref: MutableRefObject<T | null>) => JSX.Element | null
+  menuProps?: Omit<MenuProps, 'children'> & { children?: React.ReactNode }
+  portalProps?: Omit<PortalProps, 'children'> & { children?: React.ReactNode }
+  menuButtonProps?: MenuButtonProps
+}
+
+/**
+ * From https://github.com/lukasbach/chakra-ui-contextmenu
+ * @deprecated Doesn't integrate well with React Flow (keyboard navigation) and ChakraUI (focus on first element)
+ */
+export function ContextMenu<T extends HTMLElement = HTMLElement>(props: ContextMenuProps<T>) {
+  const [isOpen, setIsOpen] = useState(false)
+  const [isRendered, setIsRendered] = useState(false)
+  const [isDeferredOpen, setIsDeferredOpen] = useState(false)
+  const [position, setPosition] = useState<[number, number]>([0, 0])
+  const targetRef = useRef<T>(null)
+
+  useEffect(() => {
+    if (isOpen) {
+      setTimeout(() => {
+        setIsRendered(true)
+        setTimeout(() => {
+          setIsDeferredOpen(true)
+        })
+      })
+    } else {
+      setIsDeferredOpen(false)
+      const timeout = setTimeout(() => {
+        setIsRendered(isOpen)
+      }, 1000)
```

Comment:
My only suggestion would be to define constant variables somewhere for timeouts. They could even become user preferences one day.